zerver/actions/scheduled_messages.py

```diff
@@ -64,8 +69,8 @@ def check_schedule_message(
         # Legacy default: a scheduled message you sent from a non-API client is
         # automatically marked as read for yourself, unless it was sent to
         # yourself only.
-        read_by_sender = (
-            client.default_read_by_sender() and send_request.message.recipient != sender.recipient
+        read_by_sender = client.default_read_by_sender() and not is_message_to_self(
+            send_request.message
```

Comment:
This should be possible to compute using `addressee` metadata without fetching the `display_recipient`, which costs a cache query. `address.user_profiles()` is a list of the users involved.

Reply:
Good call. Added the `is_message_to_self` method to `addressee`. I had to pass `sender` to the utility method, since it wasn't included in `addressee.user_profiles()`.

I kept the new `is_message_to_self` function in `lib/message.py` since I'm using that in one of the tests.